### PR TITLE
Nav

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

File: themes/qgrid/layout/partials/nav.ejs [hunk 1]
```diff
@@ -1,89 +1,27 @@
...
-            <a href="#">How to install</a>
-        </li>
-        <li>
-            <a href="#">Concepts and architecture</a>
-        </li>
-        <li>
-            <a href="#">Grid model and service</a>
-        </li>
-    </ul>
+        <% var groupOrder = ['Getting started', 'API', 'Features'];
```
Comment (klumba12) [inline themes/qgrid/layout/partials/nav.ejs]: consider to have next renamings:

groupOrder -> groups
groupped -> pageGroups

File: themes/qgrid/layout/partials/nav.ejs [hunk 2]
```diff
@@ -1,89 +1,27 @@
...
+        var groupped = site.pages.reduce(function (groups, p) {
+            var key = p.group;
+            (groups[key] = groups[key] || []).push(p);
+            return groups;
```
Comment (klumba12) [inline themes/qgrid/layout/partials/nav.ejs]: usually we always use `memo` name as an aggregation result

File: themes/qgrid/layout/partials/nav.ejs [hunk 3]
```diff
@@ -1,89 +1,27 @@
...
-            <a href="#">Mobile device support</a>
-        </li>
-        <li>
-            <a href="#">Localization</a>
-        </li>
-        <li>
-            <a href="#">Context menu</a>
-        </li>
-    </ul>
+        for (var g = 0; g < groupOrder.length; g++) {
```
Comment (klumba12) [inline themes/qgrid/layout/partials/nav.ejs]: g -> i
key -> group

File: themes/qgrid/layout/partials/nav.ejs [hunk 4]
```diff
@@ -1,89 +1,27 @@
...
-            <a href="#">Another point</a>
-        </li>
-        <li>
-            <a href="#">Third unit</a>
-        </li>
-        <li>
-            <a href="#">And so on</a>
+        if (groupped.hasOwnProperty(key)){ %>
+        <li><h3><%- key %></h3></li>
+        <% var sorted = groupped[key].sort(function (a, b) {
```
Comment (klumba12) [inline themes/qgrid/layout/partials/nav.ejs]: sorted -> pages
add variable `var page = groupped[key] `

Comment (klumba12) [reply themes/qgrid/layout/partials/nav.ejs]: do sort requires to return boolean or `-1, 0, 1` ? (a.order - b.order) ?

File: themes/qgrid/source/js/menu.js [hunk 5]
```diff
@@ -25,10 +25,12 @@ function filter(e) {
 
 	for (var i = 0; i < li.length; i++) {
 		var a = li[i].getElementsByTagName('a')[0];
-		if (a.innerHTML.toLowerCase().indexOf(filter) > -1) {
-			li[i].style.display = '';
-		} else {
-			li[i].style.display = 'none';
+		if (a) {
+			if (a.innerHTML.toLowerCase().indexOf(filter) > -1) {
```
Comment (klumba12) [inline themes/qgrid/source/js/menu.js]: can we use text not html?

File: themes/qgrid/layout/partials/nav.ejs (hunk 4 above)
Comment (etokareva) [reply themes/qgrid/layout/partials/nav.ejs]: boolean

File: themes/qgrid/layout/partials/nav.ejs (hunk 2 above)
Comment (etokareva) [reply themes/qgrid/layout/partials/nav.ejs]: I'll rename, but prefer more significant names

**Review by klumba12 — APPROVED** [review]

**Pull request merged**